### ISSUE TITLE
Windows Shortcuts Issues (with OpenFilePath?)

### ISSUE THREAD

**Issue by IntrepidWanderer** [sandboxie-plus/Sandboxie#3610]
### Describe what you noticed and did

1. Put installer inside folder in sandbox. Run installer and installed a program (e.g. FirefoxPortable) into a folder C:\OtherPrograms under a sandbox that has OpenFilePath enabled.
2. Create a shortcut in Windows pointing to the program with the target line (nothing entered in "start in" line. Also Windows apparently has no issue finding the program when I browse to it as the location for the icon to be used for the shortcut):
`"C:\Program Files\Sandboxie-Plus\SandMan.exe" /box:ProgramLUse "C:\Sandboxes\ProgramLUse\drive\C\OtherPrograms\FirefoxPortable\FirefoxPortable.exe"`
![Screenshot Sandboxie FFPortable Shortcut Properties](https://github.com/sandboxie-plus/Sandboxie/assets/20574569/7d1a2e00-f872-466e-bd86-5abfbaffadce)
3. Attempt to start said program by running/double clicking shortcut. Windows pops up an error saying

Sandboxie Start

Could not invoke program:

"C:\Sandboxes\ProgramLUse\drive\C\OtherPrograms\FirefoxPortable\FirefoxPortable.exe"

System Error Code:

The system cannot find the file specified.
 (2) 

4. The above happens with any programs, including if I copy and move a program directly into the folder inside the sandbox, or with programs that have installers which automatically creates a desktop/start menu shortcut. I have Sandboxie setting for automatically adding sandbox shortcuts to the start menu enabled, and the start menu shortcuts created by Sandboxie have the same issue for programs in the folder that has "OpenFilePath" enabled.

### How often did you encounter it so far?

Always, appears to happen for any shortcuts created for programs within the same folder of the same sandbox

### Expected behavior

I expect the Windows shortcut to successfully start up the program it's pointing to while running under Sandboxie in the sandbox specified.

### Affected program

All

### Download link

N/A

### Where is the program located?

The program is installed only inside a sandbox (NOT in the real system anyway).

### Did the program or any related process close unexpectedly?

No, not at all.

### What version of Sandboxie are you running now?

Sandboxie v1.12.9 x64

### Is it a new installation of Sandboxie?

I just updated Sandboxie from a previous version (I remember which one it is).

### Is it a regression from previous versions?

Exists in an odd way in 1.12.7 (odd because it seems to come and go, not sure what specific Sandboxie setting I changed or did to make it go away)

### In which sandbox type you have this problem?

In a standard isolation sandbox (yellow sandbox icon).

### Can you reproduce this problem on a new empty sandbox?

I can confirm it also on a new empty sandbox.

### What is your Windows edition and version?

Windows 10 Education Version	22H2 OS build	19045.3930 Experience	Windows Feature Experience Pack 1000.19053.1000.0

### In which Windows account you have this problem?

A local account (Administrator)., A Microsoft account (Administrator)., An account with UAC protection completely turned off.

### Please mention any installed security software

Microsoft Defender Antivirus

### Sandboxie.ini configuration

```shell
[GlobalSettings]

FileRootPath=C:\Sandboxes\%SANDBOX%
Template=SynapticsTouchPad
Template=Edge_Fix
Template=OfficeLicensing
Template=WindowsLive
Template=WindowsRasMan
ActivationPrompt=n
ForceDisableSeconds=60
NetworkEnableWFP=y
KeyRootPath=\REGISTRY\USER\Sandbox_%USER%_%SANDBOX%
IpcRootPath=\Sandbox\%USER%\%SANDBOX%\Session_%SESSION%

[UserSettings_029800E3]

SbieCtrl_EditConfNotify=n
SbieCtrl_EnableLogonStart=n
SbieCtrl_EnableAutoStart=y
SbieCtrl_AddDesktopIcon=n
SbieCtrl_AddQuickLaunchIcon=n
SbieCtrl_AddContextMenu=n
SbieCtrl_AddSendToMenu=y
SbieCtrl_ActiveView=40021
SbieCtrl_HideWindowNotify=n
SbieCtrl_ShortcutNotify=n
SbieCtrl_ShowWelcome=n
SbieCtrl_NextUpdateCheck=1555555555
SbieCtrl_UpdateCheckNotify=n
SbieCtrl_ExplorerWarn=n
SbieCtrl_ReloadConfNotify=n
SbieCtrl_SaveRecoverTargets=n
SbieCtrl_AutoApplySettings=n
SbieCtrl_WindowCoords=1404,681,516,367
SbieCtrl_ProcessViewColumnWidths=216,62,225
BoxDisplayOrder=DefaultBox,AdminDefaultBox,AdminTestBox,GamesInstall,GamesInstallnRun,GamesRun,GManagers,ProgramLUse
SbieCtrl_ShouldDeleteNotify=n
SbieCtrl_BoxExpandedView=AdminDefaultBox,GManagers,GamesRun,TestBox
SbieCtrl_AutoStartAgent=SandMan.exe -autorun

[ProgramLUse]
Enabled=y
ConfigLevel=10
Template=AutoRecoverIgnore
Template=LingerPrograms
Template=Kaspersky
Template=BlockPorts
Template=FileCopy
Template=SkipHook
Template=OpenBluetooth
RecoverFolder=C:\Program Files (x86)
RecoverFolder=C:\Program Files
RecoverFolder=C:\OtherPrograms
RecoverFolder=%Personal%
RecoverFolder=%Start Menu%
RecoverFolder=%Desktop%
RecoverFolder=C:\Documents and Settings
RecoverFolder=D:
RecoverFolder=F:
BoxNameTitle=y
BorderColor=#00ffff,off,6
BlockDrivers=y
BlockWinHooks=n
BlockFakeInput=y
BlockSysParam=y
BlockPassword=y
OpenPipePath=winace.exe,D:\
OpenPipePath=winace.exe,F:\Downloads\
OpenPipePath=winace.exe,%Personal%
OpenPipePath=FirefoxPortable.exe,D:\
OpenPipePath=FirefoxPortable.exe,F:\Downloads\
OpenPipePath=FirefoxPortable.exe,%Personal%
OpenPipePath=H:
OpenPipePath=G:
OpenPipePath=C:\Temp
LingerProcess=peer.exe
LingerProcess=peeradapater.exe
NeverDelete=y
NeverRemove=y
OpenFilePath=C:\OtherPrograms
```

**Comment by offhub** [comment]
You cannot install applications that you run from the sandbox folder to the location on the host using `OpenFilePath`. You should use `OpenPipePath` instead.
See:
[Resource Access table](https://sandboxie-plus.github.io/sandboxie-docs/Content/ResourceAccess.html)
[OpenPipePath](https://sandboxie-plus.github.io/sandboxie-docs/Content/OpenPipePath.html)

To run an application in the sandbox folder, simply use the direct sandbox path. (or Remove the OpenFilePath/OpenPipePath directive for that location)

```
"C:\Sandboxes\ProgramLUse\drive\C\OtherPrograms\FirefoxPortable\FirefoxPortable.exe"
```

You can use SandMan.exe or Start.exe to run an application that is located in a folder on the host.

```
"C:\Program Files\Sandboxie-Plus\SandMan.exe" /box:ProgramLUse "C:\OtherPrograms\FirefoxPortable\FirefoxPortable.exe"
```

**Comment by IntrepidWanderer** [comment]
You misunderstand what I'm trying to do with OpenFilePath. I'm aware of the differences between OpenPipePath and OpenFilePath. In fact, if I had ran the installer from outside of the sandbox, it would've installed the program to the host folder of the same name outside of the sandbox, but that is not what I want. As stated in the first step of my description, I expressly moved installer to the same name folder inside the sandbox and installed to the folder *inside* the sandbox. But I still wanted to allow host folder access to other programs running into the sandbox to be able to write directly to the host folder, IF they were outside the sandbox, which is why I set that folder to OpenFilePath instead of OpenPipePath.

In any case, all of this is not really the point, as the shortcuts should work regardless with the proper path setup in the shortcut as I have, no?

**Comment by offhub** [comment]
Try this:

```
OpenFilePath=!Start.exe,C:\OtherPrograms
```

**Comment by IntrepidWanderer** [comment]
> Please try this setting and see if it resolves the issue.
> 
> ```
> OpenFilePath=!Start.exe,C:\OtherPrograms
> ```

Yep, using OpenFilePath=!Start.exe,C:\OtherPrograms to replace my old setting OpenFilePath=C:\OtherPrograms worked.

Is this just excluding all sandboxie process from the OpenFilePath restriction?

**Comment by offhub** [comment]
Sandboxie does not perform this type of check. Therefore, all programs named `Start.exe` will also be affected by this setting.
